File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/SimpleRecipe.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.{Event, Recipe}
import examples.scala.events.OrderPlaced
import examples.scala.interactions.ShipOrder

object SimpleRecipe {
  val recipe: Recipe = Recipe("example recipe")
    .withSensoryEvent(Event[OrderPlaced])
    .withInteraction(ShipOrder.interaction)
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/SimpleRecipe.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import examples.kotlin.events.OrderPlaced
import examples.kotlin.interactions.ShipOrder

@ExperimentalDsl
object SimpleRecipe {
    val recipe = recipe("example recipe") {
        sensoryEvents {
            event<OrderPlaced>()
        }
        interaction<ShipOrder>() 
    }
}